src/providers/export/tensorFlowRecords.test.ts

```diff
@@ -0,0 +1,181 @@
+import _ from "lodash";
+import { TFRecordsJsonExportProvider,
+         ITFRecordsJsonExportOptions } from "./tensorFlowRecords";
+import { ExportAssetState } from "./exportProvider";
+import registerProviders from "../../registerProviders";
+import { ExportProviderFactory } from "./exportProviderFactory";
+import { IProject, IAssetMetadata, AssetState, IRegion, RegionType,
+         ITagMetadata, IPoint } from "../../models/applicationState";
+import MockFactory from "../../common/mockFactory";
+import axios, { AxiosResponse } from "axios";
+
+jest.mock("../../services/assetService");
+import { AssetService } from "../../services/assetService";
+
+jest.mock("../storage/localFileSystemProxy");
+import { LocalFileSystemProxy } from "../storage/localFileSystemProxy";
+
+describe("TFRecords Json Export Provider", () => {
+    const baseTestProject = MockFactory.createTestProject("Test Project");
+    baseTestProject.assets = {
+        "asset-1": MockFactory.createTestAsset("1", AssetState.Tagged),
+        "asset-2": MockFactory.createTestAsset("2", AssetState.Tagged),
+        "asset-3": MockFactory.createTestAsset("3", AssetState.Visited),
+        "asset-4": MockFactory.createTestAsset("4", AssetState.NotVisited),
+    };
+    baseTestProject.sourceConnection = MockFactory.createTestConnection("test", "localFileSystemProxy");
+    baseTestProject.targetConnection = MockFactory.createTestConnection("test", "localFileSystemProxy");
+
+    const tagLengthInPbtxt = 31;
+
+    axios.get = jest.fn((url, config) => {
+        return Promise.resolve<AxiosResponse>({
+            config,
+            headers: null,
+            status: 200,
+            statusText: "OK",
+            data: [1, 2, 3],
+        });
+    });
+
+    beforeEach(() => {
+        registerProviders();
+    });
+
+    it("Is defined", () => {
+        expect(TFRecordsJsonExportProvider).toBeDefined();
+    });
+
+    it("Can be instantiated through the factory", () => {
+        const options: ITFRecordsJsonExportOptions = {
+            assetState: ExportAssetState.All,
+        };
+        const exportProvider = ExportProviderFactory.create("tensorFlowRecords", baseTestProject, options);
+        expect(exportProvider).not.toBeNull();
+        expect(exportProvider).toBeInstanceOf(TFRecordsJsonExportProvider);
+    });
+
+    describe("Export variations", () => {
+        beforeEach(() => {
+            const assetServiceMock = AssetService as jest.Mocked<typeof AssetService>;
+            assetServiceMock.prototype.getAssetMetadata = jest.fn((asset) => {
+                const mockTag: ITagMetadata = {
+                    name: "Tag 1",
```

Comment:
Not for this PR, but it might be a good idea to add these to the mock factory at some point